catalog/task/update-infra-deployments/0.1/update-infra-deployments.yaml

```diff
@@ -0,0 +1,332 @@
+        # 'private-key' - private key for Github app
+        secretName: $(params.sharedSecret)
+    - name: shared-dir
+      emptyDir: {}
+
+  steps:
+    - name: git-clone-infra-deployments
+      image: $(params.gitImage)
+      volumeMounts:
+        - name: shared-dir
+          mountPath: /shared
+      workingDir: /shared
+      env:
+        - name: TARGET_GH_REPO
+          value: "$(params.targetGHRepo)"
+      script: |
+        git clone "https://github.com/${TARGET_GH_REPO}.git" cloned
+    - name: run-update-script
+      image: $(params.scriptImage)
+      volumeMounts:
+        - name: shared-dir
+          mountPath: /shared
+      workingDir: /shared
+      script: |
+        cd cloned
+        
+        echo "snapshot:"
+        cat "$(params.snapshotPath)"
+        echo ""
+        
+        # we assume we only have one component in the service.
+        #
+        revision=$(cat "$(params.snapshotPath)" | jq -r .components[0].source.git.revision)
+        echo "revision: $revision"
+        echo $revision > ../revision.txt
+
+        originRepo=$(cat "$(params.snapshotPath)" | jq -r .components[0].source.git.url)
+        echo "origin repo: $originRepo"
+        echo $originRepo > ../originRepo.txt
+
+        # get SCRIPT from ExtraData
+        echo "extraData: $(params.extraDataJsonPath)"
+        SCRIPT=$(cat "$(params.extraDataJsonPath)" | jq -r '."infra-deployment-update-script"')
+
+        if [ "${SCRIPT}" == "" ] ; then
+          echo "No script provided via 'infra-deployment-update-script' key in extraData"
+          exit 0
+        fi
+
+        echo "original script:"
+        echo "$SCRIPT"
+        PATCHED_SCRIPT=$(echo "$SCRIPT" | sed "s/{{ revision }}/$revision/g")
+        echo "patched script:"
+        echo "$PATCHED_SCRIPT"
+        echo "$PATCHED_SCRIPT" | sh
+    - name: get-diff-files
+      image: $(params.gitImage)
+      volumeMounts:
+        - name: shared-dir
+          mountPath: /shared
+      workingDir: /shared
+      script: |
+        cd cloned
+        git status -s --porcelain | cut -c4- > ../updated_files.txt
+    # Based on https://github.com/tektoncd/catalog/tree/main/task/github-app-token/0.2/
+    - name: create-mr
+      image: >-
+       quay.io/redhat-appstudio/github-app-token@sha256:b4f2af12e9beea68055995ccdbdb86cfe1be97688c618117e5da2243dc1da18e
+      volumeMounts:
+        - name: infra-deployments-pr-creator
+          mountPath: /secrets/deploy-key
+        - name: shared-dir
+          mountPath: /shared
+      workingDir: /shared
+      env:
+        - name: GITHUBAPP_KEY_PATH
+          value: /secrets/deploy-key/private-key
+        - name: GITHUBAPP_APP_ID
+          value: "$(params.githubAppID)"
+        - name: GITHUBAPP_INSTALLATION_ID
+          value: "$(params.githubAppInstallationID)"
+        - name: GITHUB_API_URL
+          value: https://api.github.com
+        - name: TARGET_GH_REPO
+          value: "$(params.targetGHRepo)"
+      script: |
```

Comment:
This would fit really nicely in our utils instead of in the task. But, idk how much we want to remain like the location you got this from

Reply:
I'd prefer to leave this for a later PR